extensions/guacamole-auth-radius/src/main/java/org/apache/guacamole/auth/radius/RadiusConnectionService.java

```diff
@@ -0,0 +1,394 @@
+            logger.debug("Failed to init RADIUS client.", e);
+            return;
+        }
+        catch (UnknownHostException e) {
+            logger.error("Unable to resolve host: {}", e.getMessage());
+            logger.debug("Failed to resolve host.", e);
+            return;
+        }
+        catch (IOException e) {
+            logger.error("Unable to communicate with host: {}", e.getMessage());
+            logger.debug("Failed to communicate with host.", e);
+            return;
+        }
+
+    }
+
+    /**
+     * Creates a new instance of RadiusAuthentictor, configured with
+     * parameters specified within guacamole.properties.
+     *
+     * @return
+     *     A new RadiusAuthenticator instance which has been configured
+     *     with parameters from guacamole.properties, or null if
+     *     configuration fails.
+     *
+     */
+    private RadiusAuthenticator setupRadiusAuthenticator() {
+
+        // If we don't have a radiusClient object, yet, don't go any further.
+        if (radiusClient == null) {
+            logger.error("RADIUS client hasn't been set up, yet.");
+            logger.debug("We can't run this method until the RADIUS client has been set up.");
+            return null;
+        }
+
+        String radAuthName;
+        String caFile;
+        String caPassword;
+        String caType;
+        String keyFile;
+        String keyPassword;
+        String keyType;
+        Boolean trustAll;
+        String innerProtocol;
+        LocalEnvironment guacEnv;
+        String basePath;
+
+        // Pull configuration parameters from guacamole.properties
+        try {
+            guacEnv = new LocalEnvironment();
+            basePath = guacEnv.getGuacamoleHome().getAbsolutePath() + '/';
+            radAuthName = confService.getRadiusAuthProtocol();
+            caFile = confService.getRadiusCAFile();
+            caPassword = confService.getRadiusCAPassword();
+            caType = confService.getRadiusCAType();
+            keyFile = confService.getRadiusKeyFile();
+            keyPassword = confService.getRadiusKeyPassword();
+            keyType = confService.getRadiusKeyType();
+            trustAll = confService.getRadiusTrustAll();
+            innerProtocol = confService.getRadiusEAPTTLSInnerProtocol();
+            
+        }
+        catch (GuacamoleException e) {
+            logger.error("Error retrieving configuration.");
+            logger.debug("Error getting config parameters from file.");
+            return null;
+        }
+
+        RadiusAuthenticator radAuth = radiusClient.getAuthProtocol(radAuthName);
+        if (radAuth == null)
+            return null;
+
+        // If we're using any of the TLS protocols, we need to configure them
+        if (radAuth instanceof PEAPAuthenticator || 
+            radAuth instanceof EAPTLSAuthenticator || 
+            radAuth instanceof EAPTTLSAuthenticator) {
+
+            if (caFile != null && !caFile.isEmpty())
```

Comment:
So, this sort of logic (repeated throughout this section) seems like the sort of thing that should be left to the code handling the property contents. (Stay [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself), right?)

Normally, we handle the empty string as a perfectly legitimate value for a property in `guacamole.properties`. If a user specifies the empty string for a property, but that value is *not* legitimate, then the proper thing would be to error out, not ignore the value entirely.

In this case, I'd say simply assume that the value is non-empty and treat it as a filename. Ideally, encapsulate the value within a `File` and leave the parsing up to the configuration service. That way things are either `null` (not specified) or a value. Period.